### PR TITLE
ESC-50

### PR DESCRIPTION
**Pull request by justinsalamon**
This PR adds the ESC-50 dataset to `soundata`. It is structured very similarly to urbansound8k so should be an easy lift.

### CODE REVIEW

**Comment by justinsalamon** [comment]
### Adding loader for ESC-50

### Description

Please include the following information at the top level docstring for the dataset's module mydataset.py:

- [x] Describe annotations included in the dataset
- [x] Indicate the total duration of the dataset in hours, and (optionally) also list the number of individual files
- [x] Mention the origin of the dataset (e.g. creator, institution)
- [x] Describe the type of audio included in the dataset
- [x] Indicate any relevant papers related to the dataset
- [x] Include a description about how the data can be accessed and the license it uses (if applicable)

#### Dataset loaders checklist:

- [x] Create a script in `scripts/`, e.g. `make_my_dataset_index.py`, which generates an index file.
- [x] Run the script on the canonical version of the dataset and save the index in `soundata/indexes/` e.g. `my_dataset_index.json`.
- [x] Create a module in soundata, e.g. `soundata/my_dataset.py`
- [x] Create tests for your loader in `tests/`, e.g. `test_my_dataset.py`
- [x] Add your module to `docs/source/soundata.rst` and `docs/source/quick_reference.rst`
- [x] Run `tests/test_full_dataset.py` on your dataset.

If your dataset **is not fully downloadable** there are two extra steps you should follow:
- [ ] Contacting the soundata organizers by opening an issue or PR so we can discuss how to proceed with the closed dataset.
- [ ] Show that the version used to create the checksum is the "canonical" one, either by getting the version from the dataset creator, or by verifying equivalence with several other copies of the dataset.
- [ ] Make sure someone has run `pytest -s tests/test_full_dataset.py --local --dataset my_dataset` once on your dataset locally and confirmed it passes

#### Please-do-not-edit flag
To reduce friction, we will make commits on top of contributor's pull requests by default unless they use the `please-do-not-edit` flag. If you don't want this to happen don't forget to add the flag when you start your pull request.

**Comment by justinsalamon** [comment]
@magdalenafuentes ready for CR!

File: docs/source/overview.rst [hunk 1]
```diff
@@ -21,14 +21,13 @@ into a standard format to be used for experimenting or evaluating. ``soundata``
 
     print(soundata.list_datasets())
 
-    tinysol = soundata.initialize('tinysol')
-    tinysol.download()
+    urbansound8k = soundata.initialize('urbansound8k')
```
Comment (magdalenafuentes) [inline docs/source/overview.rst]: nit^2: call it `dataset`? so people get the idea you can iterate in `soundata.DATASETS` and use the same code for everything ^^

File: docs/source/table.rst [hunk 2]
```diff
@@ -5,274 +5,21 @@
    * - Dataset
      - Downloadable?
      - Annotation Types
-     - Tracks
+     - Clips
```
Comment (magdalenafuentes) [inline docs/source/table.rst:8]: Duration (hs)? :)

File: scripts/make_esc50_index.py [hunk 3]
```diff
@@ -0,0 +1,52 @@
+import argparse
+import hashlib
+import json
+import os
+import glob
+from soundata.validate import md5
```
Comment (magdalenafuentes) [inline scripts/make_esc50_index.py:6]: ♥️

File: soundata/datasets/esc50.py [hunk 4]
```diff
@@ -0,0 +1,261 @@
...
+
+import librosa
+import numpy as np
+import csv
+
+from soundata import download_utils
+from soundata import jams_utils
+from soundata import core
+from soundata import annotations
+from soundata import io
```
Comment (magdalenafuentes) [inline soundata/datasets/esc50.py]: nit: one line?

File: tests/datasets/test_esc50.py [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+    audio, sr = esc50.load_audio(audio_path)
+    assert sr == 44100
+    assert type(audio) is np.ndarray
+    assert len(audio.shape) == 1  # check audio is loaded as mono
+    assert audio.shape[0] == 44100  # Check audio duration in sampels is as expected
+
+
+def test_to_jams():
+
+    # Note: original file is 5 sec, but for testing we've trimmed it to 1 sec
```
Comment (magdalenafuentes) [inline tests/datasets/test_esc50.py:52]: ♥️

**Review by magdalenafuentes — APPROVED** [review]
Beautiful loader! Just two very minor nits and ready to go!

File: docs/source/overview.rst (hunk 1 above)
Comment (justinsalamon) [reply docs/source/overview.rst]: sure, updated

File: docs/source/table.rst (hunk 2 above)
Comment (justinsalamon) [reply docs/source/table.rst:8]: I've added another column for duration in hours

File: soundata/datasets/esc50.py (hunk 4 above)
Comment (justinsalamon) [reply soundata/datasets/esc50.py]: I just did a quick search and found this: https://stackoverflow.com/a/15011456/2007700

Basically according to PEP8 importing whole packages should happen on separate lines, and importing modules from the same package should happen on one line.

So yes, I'll consolidate these into one line!

**Pull request merged**